docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,340 @@
+- and a list of `items` as an array of objects.
+
+Each object in the `items` array can contain the following parameters:
+
+<!-- TODO: fix the table content as the parameters given in the [example from the RFC](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#example) and the [TS interface](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#admin) described in the RFC are inconsistent 🤷  -->
+
+| Items parameter | Description                                                                                  | Type     |
+| --------------- | -------------------------------------------------------------------------------------------- | -------- |
+| `name`          | Name of the setting item<br/>Must use the `options.setting-name` format.                     | `String` |
+| `description`   | Description of the setting item to use in the Content-type Builder                           | `String` |
+| `intlLabel`     | Translation for the abel of the setting item                                                 | [`IntlObject`](https://formatjs.io/docs/react-intl/) |
+| `type`          | Type of setting (see [Setting types and specific options](#))                                 | `String` |
+
+<!-- TODO: add a super simple example with one base setting, one advanced setting, and a simple validator -->
+::: details Example: Declaring options for an example "color" custom field:
+
+```jsx
+// path: ./src/plugins/my-custom-field-plugin/strapi-admin.js
+
+register(app) {
+  app.customFields.register({
+    // …
+    options: {
+      base: [
+        {
+          sectionTitle: {
+            id: 'color-picker.color.section.format',
+            defaultMessage: 'Format',
+          },
+          items: [
+            {
+              intlLabel: {
+                id: 'color-picker.color.format.label',
+                defaultMessage: 'Color format',
+              },
+              name: 'options.format',
+              type: 'select',
+              value: 'hex',
+              options: [
+                {
+                  key: 'hex',
+                  value: 'hex',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.hex',
+                      defaultMessage: 'Hexadecimal',
+                    },
+                  },
+                },
+                {
+                  key: 'rgba',
+                  value: 'rgba',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.rgba',
+                      defaultMessage: 'RGBA',
+                    },
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      ],
+      advanced: [],
+      validator: (args) => ({
+        'color-picker': yup.object().shape({
```

Comment:
I made this suggestion above, I think we should remove the 'color-picker' key, and the schema object validation shape.